### DISCUSSION TITLE
[Show-InstallationProgress]::ExecuteAsUser add switch parameter

### DISCUSSION THREAD

**Discussion by rao-source** [PSAppDeployToolkit/PSAppDeployToolkit#814] [Q&A] [answered]
I've had to use the PSADT many times running it as part of a GPO and running in SYSTEM context. It would be great if, in some instances, I need to notify the user of changes happening, the current logged in user on the session, through the Show-InstallationProgress even though being run in the SYSTEM context, or Administrator context. Adding a switch parameter on the Show-InstallationProgress to ExecuteAsUser then also check if it's still running before closing and close-installationprogress would require similar.

Currently, I am using your own code to do the same and am planning on creating my own module to include in my deployments but offering as a suggestion.

Just a thought.

**Comment by mjr4077au** [comment, marked as answer]
What you're after will become quite feasible as part of PSAppDeployToolkit 4.1.0 as the way we run the UI components will change drastically. If you need any help after 4.1.0 releases (Q2 2025), please raise a new discussion so we can further look at things.
